internal/services/v1/permissions.go

```diff
@@ -105,9 +105,14 @@ func (ps *permissionServer) CheckPermission(ctx context.Context, req *v1.CheckPe
 		return nil, rewritePermissionsError(ctx, err)
 	}
 
+	// TODO(jschorr): Support caveats here
 	permissionship := v1.CheckPermissionResponse_PERMISSIONSHIP_NO_PERMISSION
-	if found, ok := cr.ResultsByResourceId[req.Resource.ObjectId]; ok && found.Membership == dispatch.DispatchCheckResponse_MEMBER {
-		permissionship = v1.CheckPermissionResponse_PERMISSIONSHIP_HAS_PERMISSION
+	if found, ok := cr.ResultsByResourceId[req.Resource.ObjectId]; ok {
+		if found.Membership == dispatch.ResourceCheckResult_MEMBER {
```

Comment:
aside from `CAVEATED_MEMBER` it also includes `NOT_MEMBER` and `UNKNOWN`, yet the code assumes that if it's not member, then it's a caveated member. Wouldn't this lead to panicking on `NOT_MEMBER`, or mislead us if `UNKNOWN` happened due to a bug?

Reply:
We don't include the entry in the `ResultsByResourceId` if it is `NOT_MEMBER`, and `UNKNOWN` should never be hit (if it is, it is a bug)

Reply:
huh, I see, so when is `NOT_MEMBER` used?

Reply:
Its used in the dispatcher internally, but not placed into the results set for efficiency reasons